http.go

```diff
@@ -361,6 +370,12 @@ func httpPutViolationsInner(w http.ResponseWriter, r *http.Request, typestr stri
 			return
 		}
 
+		// fail open if encounter an invalid type
+		if err := validateType(typestr, v.Object); err != nil {
```

Comment:
let's move this into `httpPutViolations()` so it's consistent with `httpPutViolation()`

Reply:
Eh, on second thought, I think this is fine. No need to loop through all the violations twice.